rosapi/src/rosapi/params.py

```diff
@@ -128,47 +118,93 @@ def _set_param(node_name, name, value, parameter_type=None):
         parameter.value = ParameterValue()
         parameter.value.type = parameter_type
         if parameter_type != ParameterType.PARAMETER_NOT_SET:
-            setattr(parameter.value, _parameter_type_mapping[parameter_type])
+            setattr(parameter.value, _parameter_type_mapping[parameter_type], loads(value))
 
-    try:
-        # call_get_parameters will fail if node does not exist.
-        call_set_parameters(node=_node, node_name=node_name, parameters=[parameter])
-    except Exception:
-        pass
+    assert _node is not None
+    client = _node.create_client(
+        SetParameters,
+        f"{node_name}/set_parameters",
+        callback_group=MutuallyExclusiveCallbackGroup(),
```

Comment:
Did you also want this to be reentrant callback group?

Reply:
No need for service clients as they are created only for a single service call. In fact, if I understand it correctly, using the default callback group should be also ok here but, just in case I'm wrong, I create unique callback groups for each client.